datafusion/optimizer/src/extract_equijoin_predicate.rs

```diff
@@ -387,4 +357,33 @@ mod tests {
 
         assert_plan_eq(&plan, expected)
     }
+
+    #[test]
+    fn join_with_alias_filter() -> Result<()> {
```

Comment:
👍

Reply:
BTW, I recommend add a integration-test to show the plan after all rule optimize it.

Reply:
Since we can't create a join whose condition is an alias in sql api, I add a test case with dataframe api.